### PR TITLE
docs and tests

### PR DESCRIPTION
**Pull request by mvanroon**
My apologies for putting all of this work in a single PR. Once I started working on this library, I encountered so many things that creating and merging separate PR's would've been a nightmare.

## Docs
 - Added jsdocs for all functions
 - Updated readme to use the new `applyAuthTokenInterceptor`
 - Simplified code examples

## Refactor
 - Split up `processQueue` into `resolveQueue` and `declineQueue` to make the function simpler, more predictable and easier to test
 - Renamed some functions and variables (e.g. `res` -> `newToken`, `expin` -> `expiresIn` )
 - Introduced `STORAGE_KEY` constant

## Package.json
 - Updated prepare script in package.json to use npm instead of yarn (since this project seems to use npm rather than yarn)
 - Installed missing dev dependencies (`eslint`+plugins and `prettier`)
 - Introduced `jest` to the project

## Eslint
 - The eslint config seems to have been copied from someone's React project. I cleaned it up and fixed some linting issues that this cleanup introduced

## Github actions
 - Updated github build action to use npm instead of yarn
 - Added github lint action so that prettier and eslint rules are enforced
 - Added github test action so that tests are ran on every push

## Tests
- Added unit tests for both exported functions

### CODE REVIEW

File: src/index.ts
Comment (revmischa) [inline src/index.ts:80]: It was named "expires in" because it's not a date it's how many seconds until the token is expired

Comment (mvanroon) [reply src/index.ts:80]: right, how about I rename the variable to `expiresIn` then?
I'll also rename `getExpiration` to `getExpiresIn` to make it more consistent

Comment (revmischa) [reply src/index.ts:80]: Sure

File: .eslintrc.js
Comment (mvanroon) [inline .eslintrc.js:8]: in some cases, we need to explicitly provide a fake/bogus parameter to a function to test that it fails, that's why I set the eslint config to allow casting to the 'any' type in tests files

File: .eslintrc.js
Comment (mvanroon) [inline .eslintrc.js:16]: Please let me know if you'd like to keep any of these overrides. All code is now completely eslint-compliant after removing these exceptions. I assumed this config file was simply copy pasted from another (react) project.

File: .github/workflows/build.yml
Comment (mvanroon) [inline .github/workflows/build.yml:2]: I figured code might as well be tested and linted on any push regardless of branch. Let me know if you disagree.

File: .github/workflows/build.yml
Comment (mvanroon) [inline .github/workflows/build.yml:21]: I saw a package-lock rather than a yarn.lock file, so I updated this accordingly

File: .github/workflows/lint.yml
Comment (mvanroon) [inline .github/workflows/lint.yml:19]: throws an error if the code is not properly formatted

File: .github/workflows/lint.yml
Comment (mvanroon) [inline .github/workflows/lint.yml:22]: throws an error if the code eslint one or more warnings

File: package.json
Comment (mvanroon) [inline package.json:28]: Not sure why this was in here, all it does is slow down pipelines. Let me know if you disagree.

File: tests/applyAuthTokenInterceptor.test.ts
Comment (mvanroon) [inline tests/applyAuthTokenInterceptor.test.ts:15]: this line can be removed

**Comment by revmischa** [comment]
terrific!

**Comment by mvanroon** [comment]
@revmischa before you merge and publish this, let me just quickly add a couple more tests and make sure the library is still working

**Comment by mvanroon** [comment]
Screenshot 2021-04-10 at 16 55 27

**Comment by mvanroon** [comment]
This PR is ready to be reviewed @revmischa

File: README.md
Comment (fvukojevic) [inline README.md:57]: small nit. I prefer seeing imports with `@`, but this is `README.md` so your call.

Comment (mvanroon) [reply README.md:57]: not sure what you mean here

**Comment by fvukojevic** [comment]
I'll try to go through the whole changes, but this looks great just glancing! Great job

File: tests/authTokenInterceptor.test.ts
Comment (fvukojevic) [inline tests/authTokenInterceptor.test.ts:172]: just swap `count` -> `counting`

**Review by fvukojevic — APPROVED** [review]

File: tests/authTokenInterceptor.test.ts
Comment (mvanroon) [reply tests/authTokenInterceptor.test.ts:172]: good catch, thanks

**Review by Dizzzmas — APPROVED** [review]
LGTM 🎉

**Comment by revmischa** [comment]
What do I need to do to test this locally? I ran `npm i` and made sure I have the latest version of TS on my system but I'm getting some errors that look like TS or old TSC issues
```

    /Users/cyber/dev/jb/axios-jwt/src/index.ts:174
        return decoded ? .exp
                         ^

    SyntaxError: Unexpected token '.'

    > 1 | import { STORAGE_KEY, setAccessToken } from '../src'
        | ^
      2 |
      3 | describe('setAccessToken', () => {
      4 |   it('throws an error if there are no tokens stored', () => {

      at Runtime.createScriptFromCode (node_modules/jest-runtime/build/index.js:1350:14)
      at Object.<anonymous> (tests/setAccessToken.test.ts:1:1)
```

**Comment by mvanroon** [comment]
I'm not sure. Maybe you're on an old version of node.

**Pull request merged**
